core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -415,6 +418,34 @@ class AuthorizerIntegrationTest extends KafkaServerTestHarness {
   }
 
   @Test
+  def testPatternSubscriptionWithNoTopicAccess() {
```

Comment:
This verifies that unauthorized topics matching the regex will cause the authorization exception to be thrown, but should we also have a test case which verifies that non-matching unauthorized topics do not cause the exception to be thrown?